### ISSUE TITLE
GR training & retrieval setup: action, positive items and loss

### ISSUE THREAD

**Issue by abdoul-rz** [meta-recsys/generative-recommenders#358]
Hi, thanks a lot for sharing this great work!
I have a few questions regarding your implementation and training of GR in the Retrieval and Ranking settings as formalized in the paper.

<img width="450" height="202" alt="Image" src="https://github.com/user-attachments/assets/a90153a2-56ac-41ae-8d1c-ca9e6a8b40ef" />

I’d really appreciate if you could clarify a few points:
1. Could you share the types and number of action signals you used (e.g., ratings, clicks, etc.)?
2. How do you practically define the positive item? You mention that items with action > 0 are considered positive, but what happens when there are multiple such items?
3. During training of the retrieval model, how are negative items handled? Are they removed, or do you supervise only on positives while keeping negatives in the sequence?
4. If negatives are kept, did you study how many negatives are necessary to significantly improve model performance?

**Comment by jiaqizhai** [comment]
Hi, just saw the message. 

1/ We used action signals consistent with what the production baselines (DLRMs) used.
2/ Please refer to https://github.com/meta-recsys/generative-recommenders/issues/114
3/ For retrieval we apply a sampled softmax loss consistent with our DLRM baselines. As mentioned in Table 1, our reported results for the retrieval setting apply supervisions only on positive positions.
